plugin/src/main/java/org/opensearch/ml/breaker/ThresholdCircuitBreaker.java

```diff
@@ -5,10 +5,13 @@
 
 package org.opensearch.ml.breaker;
 
+import lombok.Data;
+
 /**
  * An abstract class for all breakers with threshold.
  * @param <T> data type of threshold
  */
+@Data
 public abstract class ThresholdCircuitBreaker<T> implements CircuitBreaker {
 
     private T threshold;
```

Comment:
should we add `volatile` modifier for this field?

Reply:
It's better to add volatile so that each thread can read the up-to-date value of this.